unstable/wlr-workspace-unstable-v1.xml

```diff
@@ -0,0 +1,226 @@
+      </description>
+      <arg name="workspace" type="object" interface="zwlr_workspace_handle_v1"/>
+    </event>
+
+    <event name="done">
+      <description summary="all information about the workspace group has been sent">
+        This event is sent after all changes in the workspace group state have been
+        sent.
+
+        This allows changes to one or more zwlr_workspace_group_handle_v1 properties to
+        be seen as atomic, even if they happen via multiple events. In particular, an
+        output moving from one workspace group to another sends an output event and an
+        output_remove event to the two zwlr_workspace_group_handle_v1 objects in
+        question. The compositor sends both done events only after both other events.
+      </description>
+    </event>
+
+    <event name="destroyed">
+      <description summary="this workspace group has been destroyed">
+        This event means the zwlr_workspace_group_handle_v1 has been destroyed. It is
+        guaranteed there won't be any more events for this
+        zwlr_workspace_group_handle_v1. The zwlr_workspace_group_handle_v1 becomes inert
+        so any requests will be ignored except the destroy request.
+      </description>
+    </event>
+
+    <request name="destroy" type="destructor">
+      <description summary="destroy the zwlr_workspace_group_handle_v1 object">
+        Destroys the zwlr_workspace_group_handle_v1 object.
+
+        This request should be called either when the client does not want to use the
+        workspace_group anymore or after the closed event to finalize the destruction of
+        the object.
+      </description>
+    </request>
+  </interface>
+
+  <interface name="zwlr_workspace_handle_v1" version="1">
+    <description summary="a workspace belonging to a workspace group">
+      A zwlr_workspace_handle_v1 object represents a a workspace belonging to a
+      workspace group.
+
+      Each workspace group has a number, conveyed to the client with the number event,
+      and a list of states, conveyed to the client with the state event. The client may
+      request the compositor to activate the workspace.
+    </description>
+
+    <event name="number">
+      <description summary="workspace number changed">
+        This event is emitted immediately after the zwlr_workspace_handle_v1 is created
+        and whenever the number of the workspace changed. Workspaces within a workspace
+        group are uniquely and contiguously numbered starting from zero.
+      </description>
+      <arg name="number" type="int"/>
+    </event>
+
+    <event name="state">
+      <description summary="the state of the workspace changed">
+        This event is emitted immediately after the zwlr_workspace_handle_v1 is created
+        and each time the workspace state changes, either because of a compositor action
+        or because of a request in this protocol.
+      </description>
+      <arg name="state" type="array"/>
+    </event>
```

Comment:
How does workspace order change atomically?

Reply:
e.g. how to updates affecting multiple workspaces occur atomically

Reply:
Thanks for reviewing!

Two types of atomicity: to make sure the client knows when the re-ordering is done, my thinking is the compositor refrains from sending a `done` event  until it has sent all the `number` events to the workspace objects whose numbers would have to change as a result of moving one workspace.

The other type of atomicity, where the client doesn't want the compositor to act until it's done sending a bunch of events, I haven't thought of yet. I will need to add a request to say 'requests are done now!', I'll just check how these requests are conventionally named in other protocols.

Reply:
(Though since there are currently no requests in the protocol to move workspaces, it's not necessary yet for the second kind).